scheduler/feasible.go

```diff
@@ -465,6 +461,10 @@ func checkConstraint(ctx Context, operand string, lVal, rVal interface{}) bool {
 		return !reflect.DeepEqual(lVal, rVal)
 	case "<", "<=", ">", ">=":
 		return checkLexicalOrder(operand, lVal, rVal)
+	case structs.ConstraintAttributeIsSet:
```

Comment:
Shouldn't all the existing except `!=` and `not` require `lFound && rFound` to be true

Reply:
Otherwise this is true and that is a change in behavior and seemingly undesirable:

```
constraint {
  attribute = "${meta.my-non-existing}"
  operator  = "=="
  value     = "${meta.my-other-non-existing}"
}

Reply:
Good catch.. first I thought the existing methods for other operators already handled nils inside by returning false after casting fails. But reflect.deepequal(nil, nil) will return true, so we should capture that.

Reply:
Good catch 😬 🙈